api/src/test/java/org/openmrs/ui/framework/page/PageContextTest.java

```diff
@@ -61,4 +63,19 @@ public void getUniqueResourcesByCategory_shouldGetCorrectOrder() throws Exceptio
 		Collection<Resource> resources = pageContext.uniqueSortedResourcesByCategory(Resource.CATEGORY_JS);
 		assertThat(resources, contains(third, first, second));
 	}
+	
+	@Test
+	public void includeResource_shouldNotIncludeDuplicateResources() {
+		Resource first = new Resource(Resource.CATEGORY_JS, "source", "z_first", 1);
+		Resource second = new Resource(Resource.CATEGORY_JS, "mergepatientdata", "path", 2);
+		Resource duplicateResource = new Resource(Resource.CATEGORY_JS, "mergepatientdata", "path", 3);
+		PageContext pageContext = new PageContext(null);
+		pageContext.includeResource(first);
+		pageContext.includeResource(second);
+		pageContext.includeResource(duplicateResource);
+		
+		List<Resource> jsResourcesToIncluude = pageContext.getResourcesToInclude(Resource.CATEGORY_JS);
+		assertEquals(2, jsResourcesToIncluude.size());
+		assertEquals(3, (int) jsResourcesToIncluude.get(0).getPriority());
```

Comment:
Can you also add another assertion for the priority of the second resource?

Reply:
Its added